include/circt/Types.td

```diff
@@ -23,7 +23,7 @@ def APSIntAttr : Attr<CPred<"$_self.isa<::mlir::IntegerAttr>()">,
   let storageType = [{ ::mlir::IntegerAttr }];
   let returnType = [{ ::llvm::APSInt }];
   let constBuilderCall = "IntegerAttr::get($_builder.getContext(), $0)";
-  let convertFromStorage = "APSInt($_self.getValue(), !getType().isSigned())";
+  let convertFromStorage = "$_self.getAPSInt()";
```

Comment:
I'm a bit concerned about this change because I feel this just extracts the signedness of attribute, not type op operation. WDYT? @seldridge